content/engineering/incidents/playbooks/ci.md

```diff
@@ -0,0 +1,225 @@
+  - Take note of the failing step and go to next point.
+
+1. Is that flake related to the CI infrastructure?
+
+- The CI infrastructure often involves:
+  - Docker daemon not being reachable.
+  - Missing tools that we use to run the steps, such as `go`, `node`, `comby`, ...
+  - Errors from `asdf`, which is used to manage the above tools.
+- **Yes**: ping `@dev-experience-support` on Slack in the [#buildkite-main](https://sourcegraph.slack.com/archives/C02FLQDD3TQ) or [#dev-experience](https://sourcegraph.slack.com/archives/C01N83PS4TU) channels.
+  - If nodoby is online to help:
+    - Reach out for help in [#dev-chat](https://sourcegraph.slack.com/archives/C07KZF47K)
+    - Try [Restarting the agents](#restarting-the-agents) and restart the build.
+
+1. Is that flake related to the code:
+
+- See the process describe in the [flaky tests page](https://docs.sourcegraph.com/dev/background-information/testing_principles#flaky-tests)
+
+### Is this a failure or a flake?
+
+- Gravity: _minor_
+- Impact: Some builds will fail randomly, creating noise and slowing down the engineering team
+- Possible causes:
+  - Tests relying on timing.
+  - Race conditions.
+  - End to end tests are delicate by nature and can fail randomly due to the complexity of all involved components.
+  - State dependent test is not properly teared down and fails.
+
+#### Actions
+
+1. Immediately restart the faulty step.
+   - 💡 It will save you time while you're looking at the logs.
+   - Is the step passing now?
+     - **Yes**: See [Spotted a flake scenario](#spotted-a-flake)
+     - **No**: Give it another try, and see next point.
+1. Check on [Grafana](https://sourcegraph.grafana.net/explore?orgId=1&left=%5B%22now-12h%22,%22now%22,%22grafanacloud-sourcegraph-logs%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22buildkite%5C%22%7D%22%7D%5D) if there are any occurrences of the failures that were previously observed:
+1. Go the the "Explore" section
+1. Make sure to select `grafanacloud-sourcegraph-logs` in the dropdown at the top of page.
+1. Scope the time window to `7 Days` to make sure to find previous occurrences if there are any
+1. Enter a query such as `{app="buildkite"} |= "your error message"` where "your error message" is a string that identiy approximately the failure cause observed in the failing step.
+1. Is there a build that failed exactly like this?
+   - **Yes**:
+     1. 💡 Double check that you're looking at that the same step by inspecting the labels of message (click on the line to make them visible)
+     1. **Yes**, that's a flake. See the [Spotted a flake scenario](#spotted-a-flake)
+   - **No**: it's not a flake, reach out the team owning those tests.
+
+### Restarting the agents
+
+- Gravity: _minor_
+- Impact: May fail ongoing builds, but that's fine.
+- Possible causes:
+  - Manual restart by an engineer.
+  - Newer version of the agents needs to be deployed.
+
+#### Actions
+
+1. Open `k9s` to observe the currently running agents.
```

Comment:
Are we recommending `k9s` for everyone? (I personally have never tried it before)

Reply:
I don't think that suggests that this is the only way to go.

Either you know your way with `kubectl` and you'll grasp `k9s` in like two minutes or you don't, in which case, `k9s` will make it much simpler to interact with the cluster.

Reply:
I would suggest to use `kubectl get pods -w` 
```suggestion
1. Use`kubectl get pods -w` to observe the currently running agents (`k9s` works here too).
```